rspec-tools/rspec_tools/validation/description.py

```diff
@@ -1,55 +1,115 @@
+  return [s.replace('* ', '').strip() for s in SECTION_NAMES_FILE if s.strip()]
+
 # The list of all the sections currently accepted by the script.
 # The list includes multiple variants for each title because they all occur
 # in the migrated RSPECs.
 # Further work required to shorten the list by renaming the sections in some RSPECS
 # to keep only on version for each title.
-SECTION_NAMES_PATH = Path(__file__).parent.parent.parent.parent.joinpath('docs/section_names.adoc')
-SECTION_NAMES_FILE = SECTION_NAMES_PATH.read_text(encoding='utf-8').split('\n')
-ACCEPTED_SECTION_NAMES: Final[list[str]] = [s.replace('* ', '').strip() for s in SECTION_NAMES_FILE if s.strip()]
+ACCEPTED_ALL_SECTION_NAMES: Final[list[str]] = parse_names('rspec-tools/rspec_tools/validation/names/all_section_names.adoc')
+ACCEPTED_EDUCATION_SECTION_NAMES: Final[list[str]] = parse_names('rspec-tools/rspec_tools/validation/names/education_section_names.adoc')
+OPTIONAL_EDUCATION_SECTION_NAMES: Final[list[str]] = parse_names('rspec-tools/rspec_tools/validation/names/optional_education_section_names.adoc')
 # The list of all the framework names currently accepted by the script.
-FRAMEWORK_NAMES_PATH = Path(__file__).parent.parent.parent.parent.joinpath('docs/allowed_framework_names.adoc')
-FRAMEWORK_NAMES_FILE = FRAMEWORK_NAMES_PATH.read_text(encoding='utf-8').split('\n')
-ACCEPTED_FRAMEWORK_NAMES: Final[list[str]] = [s.replace('* ', '').strip() for s in FRAMEWORK_NAMES_FILE if s.strip()]
+ACCEPTED_FRAMEWORK_NAMES: Final[list[str]] = parse_names('rspec-tools/rspec_tools/validation/names/allowed_framework_names.adoc')
+# The list of all the "How to fix it?" subsection names accepted by the script.
+ACCEPTED_HOW_TO_FIX_IT_SUBSECTIONS_NAMES: Final[list[str]] = parse_names('rspec-tools/rspec_tools/validation/names/how_to_fix_it_subsection_names.adoc')
+# the list of all the "Resources" subsection names accepted by the script.
+ACCEPTED_RESOURCES_SUBSECTION_NAMES: Final[list[str]] = parse_names('rspec-tools/rspec_tools/validation/names/resources_subsection_names.adoc')
+
+def intersection(lst1, lst2):
+  lst3 = [value for value in lst1 if value in lst2]
+  return lst3
+def difference(lst1, lst2):
+  return list(set(lst1) - set(lst2))
 
 def validate_section_names(rule_language: LanguageSpecificRule):
   descr = rule_language.description
-  for h2 in descr.find_all('h2'):
-    name = h2.text.strip()
-    if name not in ACCEPTED_SECTION_NAMES:
-      raise RuleValidationError(f'Rule {rule_language.id} has unconventional header "{name}"')
+  h2_titles = list(map(lambda x: x.text.strip(), descr.find_all('h2')))
+
+  education_titles = intersection(h2_titles, ACCEPTED_EDUCATION_SECTION_NAMES)
+  if education_titles:
+    # we're using the progressive education format
+    missing_titles = difference(ACCEPTED_EDUCATION_SECTION_NAMES, education_titles)
+    if missing_titles and missing_titles != OPTIONAL_EDUCATION_SECTION_NAMES:
```

Comment:
As `OPTIONAL_EDUCATION_SECTION_NAMES` is a list, it might be more robust to check whether the difference between them is empty
```suggestion
    if difference(missing_titles, OPTIONAL_EDUCATION_SECTION_NAMES):
```